plugin/src/main/java/io/jenkins/plugins/coverage/model/CoverageNode.java

```diff
@@ -255,6 +255,27 @@ public void add(final CoverageLeaf leaf) {
         leaves.add(leaf);
     }
 
+    /**
+     * Removes this node from the coverage tree.
+     */
+    void remove() {
+        if (hasParent()) {
+            getParent().getChildren().remove(this);
+            clearEmptyPaths(getParent());
+        }
+    }
+
+    /**
+     * Clears an empty tree path from the bottom of the tree to the top, beginning with the passed node.
+     *
+     * @param node The {@link CoverageNode node} to begin from
+     */
+    private void clearEmptyPaths(final CoverageNode node) {
+        if (node != null && node.getChildren().isEmpty()) {
+            node.remove();
+        }
+    }
+
```

Comment:
Is there a reason why you add wrong files and remove them afterwards? Wouldn't it make more sense to handle this while constructing the tree? Then we do not make the model mutable. This is something I would like to avoid.

Reply:
Well yes, I totally agree with you.
In my opinion, the whole logic should be unnecessary and be removed after the problem with the ambiguous paths has been fixed.
I chose this option since it also relates to the reference trees, which belongs to older builds, where duplicate paths are also possible at the moment.
And even if I would simply not add files with duplicate paths, there is a possibility that the coverage tree contains empty packages afterwards without file nodes - in my opinion, there shouldn't be empty package nodes.